scripts/start-setupWorld

```diff
@@ -50,20 +46,65 @@ if [[ "$WORLD" ]] && ( isTrue "${FORCE_WORLD_COPY}" || [ ! -d "$worldDest" ] );
 
     count=$(echo "$baseDirs" | wc -l)
     if [[ $count -gt 1 ]]; then
-      baseDir="$(echo "$baseDirs" | sed -n ${WORLD_INDEX:-1}p)"
-      baseName=$(basename "$baseDir")
-      log "WARN multiple levels found, picking: $baseName"
+      baseDirsNoSpigotSuffix=$(echo "$baseDirs" | sed -re 's:(_nether|_the_end)/?$::' | sort -u)
+      if [ $(echo "$baseDirsNoSpigotSuffix" | wc -l) -eq 1 ]; then
+        baseDir="$baseDirsNoSpigotSuffix"
+        baseName=$(basename "$baseDir")
+        log "Found Spigot naming conventions, taking $baseName as main dimension"
+      else
+        baseDir="$(echo "$baseDirs" | sed -n ${WORLD_INDEX:-1}p)"
+        baseName=$(basename "$baseDir")
+        log "WARN multiple levels found, picking: $baseName"
+      fi
     elif [[ $count -gt 0 ]]; then
       baseDir="$baseDirs"
     else
       log "ERROR invalid world content"
       exit 1
     fi
+
+    if [ -d "${baseDir}_nether/DIM-1" ]; then
+      if [ -d "$baseDir/DIM-1" ]; then
+        log "WARN found Nether dimension in both $baseDir and ${baseDir}_nether, picking ${baseDir}_nether"
+        rm -r "$baseDir/DIM-1"
+      fi
+    fi
+    if [ -d "${baseDir}_the_end/DIM1" ]; then
+      if [ -d "$baseDir/DIM1" ]; then
+        log "WARN found End dimension in both $baseDir and ${baseDir}_the_end, picking ${baseDir}_the_end"
+        rm -r "$baseDir/DIM1"
+      fi
+    fi
+
+    log "Copying world..."
     rsync --remove-source-files --recursive --delete "$baseDir/" "$worldDest"
+
     if [ "$FAMILY" = "SPIGOT" ]; then
-      log "Copying end and nether ..."
-      [ -d "${baseDir}_nether" ] && rsync --remove-source-files --recursive --delete "${baseDir}_nether/" "${worldDest}_nether"
-      [ -d "${baseDir}_the_end" ] && rsync --remove-source-files --recursive --delete "${baseDir}_the_end/" "${worldDest}_the_end"
+      if [ -d "${baseDir}_nether" ]; then
+        log "Copying Spigot Nether..."
+        rsync --remove-source-files --recursive --delete "${baseDir}_nether/" "${worldDest}_nether"
```

Comment:
Looks great, curious if we couldn't make all these rsync's a single method that takes in "${baseDir}_nether" and "${worldDest}_nether". It really may clean up these lines 82 - 107. Either way this is great stuff and it doesn't need to happen, just thought it may make it easier to read and troubleshoot.

Reply:
Just an FYI, I am gradually converting rsync usage over to `sync` subcommand of mc-image-helper, which takes a subset of the arguments here

https://github.com/itzg/mc-image-helper#sync-and-interpolate